module/applications/api/dialog.mjs

```diff
@@ -0,0 +1,93 @@
+import Application5e from "./application.mjs";
+
+/**
+ * Application for creating dnd5e dialogs.
+ */
+export default class Dialog5e extends Application5e {
+  /** @override */
+  static DEFAULT_OPTIONS = {
+    tag: "dialog",
+    window: {
+      contentTag: "form",
+      minimizable: false
+    }
+  };
+
+  /** @override */
+  static PARTS = {
+    content: {
+      template: ""
+    },
+    footer: {
+      template: "templates/generic/form-footer.hbs"
+    }
+  };
+
+  /* -------------------------------------------- */
+  /*  Properties                                  */
+  /* -------------------------------------------- */
+
+  /**
+   * Form element within the dialog.
+   * @type {HTMLFormElement|void}
+   */
+  get form() {
+    return this.options.tag === "form" ? this.element : this.element.querySelector("form");
+  }
+
+  /* -------------------------------------------- */
+  /*  Rendering                                   */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _preparePartContext(partId, context, options) {
+    context = foundry.utils.deepClone(await super._preparePartContext(partId, context, options));
```

Comment:
Is there a particular need for `deepClone`?

Reply:
Because core's `HandlebarsApplicationMixin` passes the same object into `_preparePartContext` for each call, it is possible for data from one part to pollute another part, so I've been adding this to avoid that problem.

Reply:
Seems rather expensive to deepClone the entire config object (in `Application5e#_prepareContext`) each time but it's probably negligible.

Reply:
We could probably get away with creating a shallow clone, since the top-level keys are the only place I was really getting a problem.